Content.Shared/Lathe/SharedLatheSystem.cs

```diff
@@ -126,6 +126,25 @@ public string GetRecipeName(LatheRecipePrototype proto)
         return string.Empty;
     }
 
+    /// <summary>
+    /// Get the resulting item made by this recipe.
+    /// Returns null if this recipe is for creating reagents.
+    /// </summary>
+    [PublicAPI]
+    public EntProtoId? GetResult(LatheRecipePrototype proto)
+    {
+        if (proto.ResultOverride is {} result)
+            return result;
+
+        if (proto.ResultReagents != null)
+            return null;
+
+        // only check on debug to fail tests, don't want too much overhead in release
+        DebugTools.Assert(_proto.HasIndex<EntityPrototype>(proto.ID), $"Lathe recipe {proto.ID} didn't specify a result and there is no entity with the same ID!");
+
+        return proto.ID;
```

Comment:
This is comically sinful and definitely shouldnt be supported. Saving 1 line of yaml in exchange for goofy obfuscation is not worth it at all

Reply:
1 line of yaml, multipied by almost every lathe recipe (hundreds of lines saved).
if you get confused and do it wrong it will error on startup anyway so you know what happened and how to fix it

Reply:
there's no point to reduce trivial amounts of yaml when it harms the readability, loses the proper validation that the old system had, and is just a dumb use of another field. Using a separate prototype's ID as a fallback for something else is a complete anti-pattern that we (rightly) don't use anywhere else.